src/test/regress/sql/failure_real_time_select.sql

```diff
@@ -0,0 +1,102 @@
+-- 
+--  Failure tests for real time select queries
+-- 
+CREATE SCHEMA real_time_select_failure;
+SET search_path TO 'real_time_select_failure';
+SET citus.next_shard_id TO 190000;
+
+-- Preparation
+SELECT citus.mitmproxy('conn.allow()');
+SET citus.shard_count to 4;
+SET citus.shard_replication_factor to 1;
+
+-- create tables
+CREATE TABLE test_table(id int, value_1 int, value_2 int);
+SELECT create_distributed_table('test_table','id');
+
+-- Populate data to the table
+INSERT INTO test_table VALUES(1,1,1),(1,2,2),(2,1,1),(2,2,2),(3,1,1),(3,2,2);
+
+-- Kill when the first COPY command arrived, since we have a single placement 
+-- it is expected to error out.
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").kill()');
+SET client_min_messages TO ERROR;
+DO $$
+BEGIN
+	SELECT count(*) FROM test_table;
+	
+	EXCEPTION WHEN OTHERS THEN
+		IF SQLERRM LIKE 'failed to execute task%' THEN
+			RAISE 'Task failed to execute';
+		END IF;
+END;
+$$;
+
+-- Since the outer query uses the connection opened by the CTE,
+-- killing connection after first successful query should break. 
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").after(1).kill()');
```

Comment:
One more test suggestion: Maybe the same test without `after(1)` because we don't have any real-time selects within ctes that is failed in the first step?